### PR TITLE
feat: support glob patterns in ignore_paths configuration

### PR DESCRIPTION
**Pull request by mohammedahmed18**
## Summary

- Add support for glob patterns in `ignore_paths` CLI option and configuration
- Previously only literal paths were supported, now patterns like `**/*.test.js`, `*.log`, `dist/*` work
- Patterns are expanded relative to the module root directory

## Changes

- Added `is_glob_pattern()` function to detect glob pattern characters (`*`, `?`, `[`)
- Added `normalize_ignore_paths()` function that handles both literal paths and glob patterns
- Updated CLI argument processing to use the new function
- Added comprehensive unit tests (20 tests covering various patterns and edge cases)

## Usage Examples

CLI:
```bash
codeflash --ignore-paths "node_modules" "*.log" "**/*.test.js" "dist/*"
```

Config (`pyproject.toml`):
```toml
[tool.codeflash]
ignore_paths = ["node_modules", "**/*.test.js", "*.log", "dist"]
```

## Test plan

- [x] Unit tests for `is_glob_pattern()` function
- [x] Unit tests for `normalize_ignore_paths()` function covering:
  - Empty list
  - Literal existing/non-existing paths
  - Single asterisk patterns (`*.log`)
  - Double asterisk patterns (`**/*.py`)
  - Directory content patterns (`dir/*`)
  - Question mark patterns (`file?.txt`)
  - Bracket patterns (`file[12].txt`)
  - Mixed literal and patterns
  - Deduplication
  - Nested directory patterns
  - Absolute and relative paths

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: codeflash/code_utils/code_utils.py [hunk 1]
```diff
@@ -27,6 +27,57 @@
 
 BLACKLIST_ADDOPTS = ("--benchmark", "--sugar", "--codespeed", "--cov", "--profile", "--junitxml", "-n")
 
+# Characters that indicate a glob pattern
+GLOB_PATTERN_CHARS = frozenset("*?[")
+
+
+def is_glob_pattern(path_str: str) -> bool:
```
Comment (Saga4) [inline codeflash/code_utils/code_utils.py:34]: Great, Can refactor code in similar function like get_valid_subdirs as well.
